src/engine/ruleset/decoders/sonicwall/firewall-generated.yml

```diff
@@ -0,0 +1,162 @@
+      - rsa.identity.user_sid_dst: $~rsa.identity.user_sid_dst
+      - related.hosts: +a_append/$host.name
+      - related.hosts: +a_append/$source.address
+      - related.hosts: +a_append/$destination.address
+      - rsa.network.host_dst: $destination.address
+      # TODO: problem found here +a_append/$source.ip/$destination.ip
+      # For some reason if source.ip doesn't exist and destination.ip does exist, neither will be added.
+      # Refactor this decoder when that issue is resolved.
+      - related.ip: +a_append/$source.ip
+      - related.ip: +a_append/$destination.ip
+      - related.ip: +a_append/$source.nat.ip
+      - related.ip: +a_append/$destination.nat.ip
+  - check:
+      - ~log.start_message: +ef_not_exists
+    logpar:
+      - rsa.time.event_time: <~log.year>-<~log.mounth>-<~log.day>T<~>
+    map:
+      # TODO: It is not yet possible to include escaped characters in auxiliary arguments.
+      # When this functionality is implemented, add it to the decoder
+      - aux_slash: '/'
+      - rsa.time.date: +s_concat/$~log.year/$aux_slash/$~log.mounth/$aux_slash/$~log.day
+      - aux_slash: +ef_delete
+  - check:
+      - ~log.note: +ef_exists
+    logpar:
+      - ~log.note: <~log.note.header> <~log.note.protocol.header>:<~log.note.protocol.payload>
+    map:
+      - network.protocol: $~log.note.protocol.payload
+      # If the "fw_action" field is present in the log event, then event.action is mapped to the same value of "fw_action".
+      # If fw_action is not present in the log event, the content of the "msg" field is examined and it is verified that
+      # it is equal to some of the allowed values (cancel, deny, allow, accept or block).
+      # If it doesn't match any of those values and also fw_action doesn't exist, then event.action doesn't get mapped.
+  - check:
+      - ~log.fw_action: +ef_exists
+    map:
+      - event.action: $~log.fw_action
+      - rsa.internal.event_desc: $~log.msg_action
+  - check:
+      - ~log.fw_action: +ef_not_exists
+      - ~log.msg_action: accept
+    map:
+      - event.action: $~log.msg_action
+  - check:
+      - ~log.fw_action: +ef_not_exists
+      - ~log.msg_action: allow
+    map:
+      - event.action: $~log.msg_action
+  - check:
+      - ~log.fw_action: +ef_not_exists
+      - ~log.msg_action: block
+    map:
+      - event.action: $~log.msg_action
+  - check:
+      - ~log.fw_action: +ef_not_exists
+      - ~log.msg_action: cancel
+    map:
+      - event.action: $~log.msg_action
+  - check:
+      - ~log.fw_action: +ef_not_exists
+      - ~log.msg_action: deny
+    map:
+      - event.action: $~log.msg_action
+  - check: >-
+        +ef_not_exists/~log.fw_action AND ~log.msg_action!=accept AND
+        ~log.msg_action!=allow AND ~log.msg_action!=block AND
+        ~log.msg_action!=cancel AND ~log.msg_action!=deny
+    map:
+      - event.action: +ef_delete
+  - check:
+      - event.action: +ef_exists
+    map:
+      - rsa.misc.action: +a_append/$event.action
```

Comment:
You could directly try to append the value, it is not necessary to check it before

Reply:
Same in the other cases, and also when copying values